Tests/SonarScanner.MSBuild.PreProcessor.Test/EngineResolution/EngineResolverTests.cs

```diff
@@ -0,0 +1,63 @@
+﻿/*
+ * SonarScanner for .NET
+ * Copyright (C) 2016-2025 SonarSource SA
+ * mailto: info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using NSubstitute.ReturnsExtensions;
+
+namespace SonarScanner.MSBuild.PreProcessor.EngineResolution.Test;
+
+[TestClass]
+public class EngineResolverTests
+{
+    [TestMethod]
+    public async Task ResolveEngine_ReturnsLocalEnginePath_WhenEngineJarPathIsSet()
```

Comment:
```suggestion
    public async Task ResolveEngine_EngineJarPathIsSet_LocalEnginePath()
```
https://github.com/SonarSource/sonar-dotnet-enterprise/blob/master/docs/coding-style.md#unit-tests